odbc/src/api/data.rs

```diff
@@ -1077,7 +1101,7 @@ pub mod i16_len {
         let message = cstr::to_widechar_vec(message);
         let (len, ret) = set_output_wstring_helper(&message, output_ptr, buffer_len);
         // Only copy the length if the pointer is not null
-        ptr_safe_write(text_length_ptr, len as SmallInt);
+        ptr_safe_write(text_length_ptr, len.try_into().unwrap_or(SmallInt::MAX));
```

Comment:
uhm, I don't think that it makes sense to default to SmallInt::MAX.
There is something fishy here.
It looks like if the buffer size is > SmallInt::MAX and message length is > SmallInt::MAX then we will report that we copied SmallInt::MAX characters in the buffer, while we actually copied more.
Shall we try to reconcile types. Maybe changing text_length_ptr to usize?

Reply:
Everywhere that this function is called traces back up to some ODBC function (e.g., SQLDescribeColW) that has the `text_length_ptr` variable defined as a `*mut SmallInt` and the `buffer_len` variable defined as a `SmallInt` by the ODBC spec. 

For example, look at the `name_length` and `buffer_length` parameters in the below function: 
```
pub unsafe extern "C" fn SQLDescribeColW(
    hstmt: HStmt,
    col_number: USmallInt,
    col_name: *mut WideChar,
    buffer_length: SmallInt,
    name_length: *mut SmallInt,
    data_type: *mut SqlDataType,
    col_size: *mut ULen,
    decimal_digits: *mut SmallInt,
    nullable: *mut SmallInt,
)
```
I think this is why this function ended up getting defined like this even though it doesn't make sense that we have a `buffer_len` of `usize` and a `text_length_ptr` of type `SmallInt`. Furthermore, in all the places it's used, `buffer_len` starts out as a `SmallInt`, then we always cast to `usize` when passing it to this function. So, unless we plan on using this function in more places, we should probably just change `buffer_len` to a `SmallInt` and stop casting it to `usize` where we call this function.

Reply:
Yep, the max size for `text_len` we ever can get appears to be an i32 or an i16. I'm going to go through and change these, then see what updates need to be made elsewhere